templates/controllers/session.ex

```diff
@@ -44,8 +44,29 @@ defmodule <%= inspect @web_pascal_case %>.Session do
     |> Identity.delete_all_user_sessions()
 
     conn
-    |> clear_session()
+    |> renew_session()
     |> put_flash(:info, "Successfully signed out.")
     |> redirect(to: ~p"/")
   end
+
+  # This function renews the session ID and erases the whole
+  # session to avoid fixation attacks. If there is any data
+  # in the session you may want to preserve after log in/log out,
+  # you must explicitly fetch the session data before clearing
+  # and then immediately set it after clearing, for example:
+  #
+  #     defp renew_session(conn) do
+  #       preferred_locale = get_session(conn, :preferred_locale)
+  #
+  #       conn
+  #       |> configure_session(renew: true)
+  #       |> clear_session()
+  #       |> put_session(:preferred_locale, preferred_locale)
+  #     end
+  #
+  defp renew_session(conn) do
+    conn
+    |> configure_session(renew: true)
+    |> clear_session()
+  end
```

Comment:
Following the lead from Phoenix's [auth generator](https://github.com/phoenixframework/phoenix/blob/4c10bab46c4cd530240129aff682d54d2950b80b/priv/templates/phx.gen.auth/auth.ex#L63), should we also clear the CSRF token?


```suggestion
  defp renew_session(conn) do
    delete_csrf_token()
  
    conn
    |> configure_session(renew: true)
    |> clear_session()
  end
```

Reply:
Good idea. I will do this.